mathics/builtin/logic.py

```diff
@@ -116,9 +116,139 @@ class Not(PrefixOperator):
     rules = {
         'Not[True]': 'False',
         'Not[False]': 'True',
+        'Not[Not[expr_]]': 'expr',
     }
 
 
+class Implies(BinaryOperator):
+    """
+    <dl>
+    <dt>'Implies[$expr1$, $expr2$]'
+    <dt>'$expr1$ \u21D2 $expr2$'
+        <dd>evaluates each expression in turn, returning 'True'
+        as soon as the first expression evaluates to 'False'. If the
+        first expression evaluates to 'True', 'Implies' returns the
+        second expression.
+    </dl>
+
+    >> Implies[False, a]
+     = True
+    >> Implies[True, a]
+     = a
+
+    If an expression does not evaluate to 'True' or 'False', 'Implies'
+    returns a result in symbolic form:
+    >> Implies[a, Implies[b, Implies[True, c]]]
+     = a \u21D2 b \u21D2 c
+    """
+
+    operator = '\u21D2'
+    precedence = 200
+    attributes = ('HoldAll', 'OneIdentity')
```

Comment:
Thanks a lot. I didn't know about Attributes function... I will correct it and the following problems.